employee_scheduling.py

```diff
@@ -152,73 +152,49 @@ def run_cqm(cqm):
     feasible_sampleset = sampleset.filter(lambda row: row.is_feasible)
 
     num_feasible = len(feasible_sampleset)
-    errors = " "
+    errors = defaultdict(list)
     if num_feasible == 0:
-        msg = "\nNo feasible solution found.\n"
-        errors += msg
-        print(msg)
         sat_array = sampleset.first.is_satisfied
 
-        # if sampleset is all 0's, set at least one variable to 1 
+        # if sampleset is all 0's, set at least one variable to 1
         # --> this is needed for "Solve CQM" button to function propertly
         s_vals = set(sampleset.first.sample.values())
         if s_vals == {0.0}:
             sampleset.first.sample[list(cqm.variables)[0]] = 1.0
 
         constraints = sampleset.info["constraint_labels"]
+
+        msgs = [
```

Comment:
If feels very fragile to rely on the order of `sampleset.info["constraint_labels"]` to be index-linked to this list. I'd rather parse the constraint label, maybe with a regex, and match it to the appropriate display name.

Reply:
I'm confused about what you're saying. It isn't looking at the order of `sampleset.info["constraint_labels"]`, each of those labels is a string that starts with a number and I take those numbers and use them as indices for the msgs list, which seems similar to your regex solution. Or am I misunderstanding what you're saying?